ibis/backends/pandas/execution/arrays.py

```diff
@@ -110,6 +110,21 @@ def execute_array_repeat_scalar(op, data, n, **kwargs):
     return np.tile(data, max(n, 0))
 
 
-@execute_node.register(ops.ArrayCollect, (pd.Series, SeriesGroupBy))
-def execute_array_collect(op, data, aggcontext=None, **kwargs):
-    return aggcontext.agg(data, np.array)
+@execute_node.register(ops.ArrayCollect, pd.Series, bool, (type(None), pd.Series))
+def execute_array_collect(op, data, distinct, where, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data.loc[where] if where is not None else data,
+        np.unique if distinct else np.array,
```

Comment:
use pd.unique as np.unique sorts

Reply:
Technically the order is undefined, but happy to use `pd.unique`.

Reply:
sure - pd.unique preserves whatever it is

Reply:
Fixed! Appreciate the review!